### PR TITLE
Add Traffic

### LINKED ISSUE #498: Using the Traffic API
Do you have anything for the traffic API?
It isn't in the docs...
https://developer.github.com/v3/repos/traffic/
Thank you!
See #548

### PR DESCRIPTION
**Pull request by m1guelpf**
Fix #498
Add support for the traffic API.
https://developer.github.com/v3/repos/traffic/

### CODE REVIEW

**Comment by Nyholm** [comment]
Excellent. Thank you for this PR. 

I see that it is missing some tests. Would you add some and I'll do a review of this.

**Comment by m1guelpf** [comment]
@Nyholm I have looked in the tests, but I haven't seen a way to make tests. I am not familiar with tests, sorry :cry: 
Anyway, you merged my other PR (#489) without providing tests (I later did in #490 )

**Comment by Nyholm** [comment]
No worries. I'll help you out tonight when I get back from work. 

Yes, merging #489 without tests was a mistake of mine. I'm happy you corrected my mistake in #490.

**Comment by m1guelpf** [comment]
@Nyholm Thank you!

**Comment by m1guelpf** [comment]
Fixed everything and checked it works!
[Screenshot](http://i.imgur.com/ckscx8W.png)

File: lib/Github/Api/Repository/Traffic.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    /**
+     * @link https://developer.github.com/v3/repos/traffic/#views
+     *
+     * @param string $owner
+     * @param string $repository
+     *
+     * @return array
+     */
+    public function views($owner, $repository)
```
Comment (Nyholm) [inline lib/Github/Api/Repository/Traffic.php]: Missing parameter `per`. See https://developer.github.com/v3/repos/traffic/#views

File: lib/Github/Api/Repository/Traffic.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    /**
+     * @link https://developer.github.com/v3/repos/traffic/#clones
+     *
+     * @param string $owner
+     * @param string $repository
+     *
+     * @return array
+     */
+    public function clones($owner, $repository)
```
Comment (Nyholm) [inline lib/Github/Api/Repository/Traffic.php]: Missing parameter `per`. See https://developer.github.com/v3/repos/traffic/#clones

**Comment by Nyholm** [comment]
I've created a PR to explain how to write tests. See #503

Is it understandable? Should I add something to it so it makes more sense?

**Comment by m1guelpf** [comment]
@Nyholm Replied in your PR.

File: test/Github/Tests/Api/Repository/TrafficTest.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Github\Tests\Api\TestCase;
+
+class TrafficTest extends TestCase
+{
+    // ...
```
Comment (Nyholm) [inline test/Github/Tests/Api/Repository/TrafficTest.php]: Should be removed

File: test/Github/Tests/Api/Repository/TrafficTest.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    /**
+     * @test
+     */
+    public function shoulddoSomething()
```
Comment (Nyholm) [inline test/Github/Tests/Api/Repository/TrafficTest.php]: Please update the name of this function

Comment (Nyholm) [inline test/Github/Tests/Api/Repository/TrafficTest.php]: You should also update the test content

**Comment by m1guelpf** [comment]
@Nyholm As I said earier, I don't know hot to write tests...

**Comment by Nyholm** [comment]
> As I said earier, I don't know hot to write tests...

Tests specific to this library or in general?

Read about how to write tests to the Github library here: https://github.com/KnpLabs/php-github-api/blob/master/doc/testing.md

Testing in general: https://www.startutorial.com/articles/view/phpunit-beginner-part-1-get-started

File: lib/Github/Api/Repository/Traffic.php (hunk 2 above)
Comment (m1guelpf) [inline lib/Github/Api/Repository/Traffic.php]: Done

Comment (Nyholm) [inline lib/Github/Api/Repository/Traffic.php]: Still not added.

**Comment by m1guelpf** [comment]
@Nyholm Oops! I deleted the repo...
Is there any way I can recover it?

**Comment by Nyholm** [comment]
I do not think so. Sorry.

**Comment by Nyholm** [comment]
I added this to the milestone 2.2. Is it possible that you can add tests to this PR this weekend?

**Comment by m1guelpf** [comment]
@Nyholm I'll try

**Comment by m1guelpf** [comment]
@Nyholm Moved to #548

**Pull request closed without merging**